modules/docs/arrow-docs/docs/docs/arrow-fx/integrations/kotlinxcoroutines/README.md

```diff
@@ -0,0 +1,154 @@
+
+The * marked ops are available within this integration module, offering different alternatives depending on the project's needs. Both options offered in this integration module are described below:
+
+## Integrating Coroutines with IO
+
+If you'd like to introduce `IO` in your project, you might want to keep using the Coroutines style of cancellation with scopes. This is specially useful on *Android* projects where the Architecture Components [already provide handy scopes for you](https://developer.android.com/topic/libraries/architecture/coroutines#lifecycle-aware).
+
+### unsafeRunScoped
+
+Runs the specific `IO` with a [CoroutineScope](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/index.html), so it will be automatically canceled when the scope does as well:
+
+
+```kotlin:ank:playground
+import arrow.fx.IO
+import arrow.unsafe
+import arrow.fx.extensions.io.unsafeRun.runBlocking
+import arrow.fx.extensions.fx
+//sampleStart
+suspend fun sayHello(): Unit =
+  println("Hello World")
+
+suspend fun sayGoodBye(): Unit =
+  println("Good bye World!")
+
+fun greet(): IO<Unit> =
+  IO.fx {
+    !effect { sayHello() }
+    !effect { sayGoodBye() }
+  }
+
+fun main() { 
+  greet().unsafeRunScoped(viewModelScope) // Android example, any other scope is possible as well.
+
+  // when viewModelScope.cancel() is invoked, it will cancel the operation above
+}
+//sampleEnd
+```
+
+
+## Alternatively, integrating IO with Coroutines
+
+Sometimes you might want to not switch the runtime of your project and slowly integrate `IO` instead. For this use case we've added some extensions to make `IO` work with the Coroutines runtime.
+
+*IMPORTANT NOTE*: The way Coroutines handle errors is by throwing exceptions after you run your operations. Because of this it's important to clarify that your operation might crash your app if you're not handling errors (in case of IO, `handleError`) or try-catching the execution.
+
+### suspendCancellable
+
+The `suspendCancellable` function will turn an `IO` into a cancellable coroutine, allowing you to cancel it within its scope like any other coroutine.
+
+```kotlin:ank:playground
+import arrow.fx.IO
+import arrow.fx.extensions.fx
+import arrow.integrations.kotlinx.suspendCancellable
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+//sampleStart
+suspend fun sayHello(): Unit =
+  println("Hello World")
+
+suspend fun sayGoodBye(): Unit =
+  println("Good bye World!")
+
+fun greet(): IO<Unit> =
+  IO.fx {
+    !effect { sayHello() }
+    !effect { sayGoodBye() }
+  }
+
+fun main() {
+  // or any other scope. This IO would stop as soon as the scope is cancelled
```

Comment:
I would modify the example to be an Android activity because the Global scope is supposed to be uncancellable.

Reply:
I ended up simply calling it `scope` to make it more agnostic